lib/actions/createEvent.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+const elasticio = require('elasticio-node');
+const messages = elasticio.messages;
+const _ = require('lodash');
+
+const Helper = require('../helper');
+const baseComponentFunction = require('../baseComponentFunction');
+const processEvent = require('../processEventDataHelper').processEvent;
+
+function getCalendars(cfg, cb) {
+
+  function processData(items) {
+    console.log("Processing calendar data ...");
+    let result = {};
+    _.forEach(items.value, function setItem(item) {
+      result[item.id] = item.name;
+    });
+    return result;
+  }
+
+  console.log("Getting calendar data ...");
+  const instance = new Helper(cfg, 'Calendars.Read');
+  return instance
+    .get('/me/calendars')
+    .then(processData)
+    .nodeify(cb);
+}
+
+
+
+function processAction(msg, cfg) {
+
+  const postRequestBody = processEvent(cfg, msg.body);
```

Comment:
As I mentioned in another comment ``processEvent`` should be a promise otherwise a lot of different errors inside this function will never be emitted as ``error`` event.